### PR TITLE
Automatic issue sync

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/tasks/BindingStorageUpdateTask.java [hunk 1]
```diff
@@ -149,40 +150,49 @@ public void doRun() {
...
+  private static void updateAllProjectIssuesForCurrentBranch(ConnectedSonarLintEngine engine, ServerConnection connection, TaskProgressMonitor monitor,
+    Collection<Project> projectsToUpdate) {
+    var allProjectAndBranchesToSync = new HashSet<ProjectBindingManager.ProjectKeyAndBranch>();
+    for (Project project : projectsToUpdate) {
+      ProjectBindingManager bindingManager = getService(project, ProjectBindingManager.class);
+      var projectAndBranchesToSync = bindingManager.getUniqueProjectKeysAndBranchesPairs();
+      allProjectAndBranchesToSync.addAll(projectAndBranchesToSync);
+    }
+    allProjectAndBranchesToSync
+      .forEach(pb -> engine.downloadAllServerIssues(connection.getEndpointParams(), connection.getHttpClient(), pb.getProjectKey(), pb.getBranchName(), monitor));
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/tasks/BindingStorageUpdateTask.java:190]: Why not use `syncServerIssues` here ? You don't want to use the incremental update but reset the storage ?

**Review by damien-urruty-sonarsource — APPROVED** [review]
LGTM

File: build.gradle.kts [hunk 2]
```diff
@@ -63,6 +63,7 @@ allprojects {
     }
 
     repositories {
+        mavenLocal()
```
[the remark this answers is not in the document]
Comment (henryju) [reply build.gradle.kts:66]: Indeed, I will remove

File: src/main/java/org/sonarlint/intellij/tasks/BindingStorageUpdateTask.java (hunk 1 above)
Comment (henryju) [reply src/main/java/org/sonarlint/intellij/tasks/BindingStorageUpdateTask.java:190]: This is bit misleading, but `syncServerIssues` is a noop for older SQ versions and SC. Here we want to refresh issues for all SQ versions.

**Pull request merged**